code/UI/Interactions/InteractionMenu.cs

```diff
@@ -28,65 +61,95 @@ public void SetEntity( Entity entity )
 		}
 	}
 
+	protected void Clear()
+	{
+		InteractionList.Clear();
+		Options.DeleteChildren( true );
+	}
+
 	public void CreatePanel()
 	{
+		int i = 0;
 		foreach ( Interaction interaction in InteractionList )
-			AddInteractionOption( interaction );
+		{
+			AddInteractionOption( interaction, i );
+			i++;
+		}
 	}
 
-	private void AddInteractionOption( Interaction interaction )
+	private void AddInteractionOption( Interaction interaction, int index )
 	{
 		if ( !interaction.CanResolve() )
 			return;
 
-		Label label = AddChild<Label>( "interaction-entry" );
-		label.Text = interaction.NiceName;
+		var label = Options.AddChild<InteractionEntry>( "interaction-entry" );
+		label.Name.Text = interaction.NiceName;
+		label.BindClass( "active-interaction", () => index == CurrentInteractionIndex );
 
-		label.AddEventListener( "onclick", () =>
-		{
-			interaction.ClientResolve();
+		UpdateArrows();
+	}
 
-			if ( interaction.ResolveOnServer )
-				Interaction.TryServerResolve( interaction.Owner.NetworkIdent, interaction.ID );
+	protected void Use()
+	{
+		var interaction = InteractionList[CurrentInteractionIndex];
```

Comment:
To follow up on above, after selecting an InteractionEntry, we clear the list, and are still trying to access index n.